### ISSUE TITLE
star ratings dont save after spotify closes

### ISSUE THREAD

**Issue by brvsket** [brimell/spicetify-star-ratings#11]
title

**Comment by dantdmdarealguy** [comment]
bump

**Comment by brimell** [comment]
Try removing the extension and re-installing it through spicetify marketplace: everything seems to be working for me so I'm not sure what I can do. (maybe provide some more info or open the dev tools console and send an ss of what is going on there?)

**Comment by brimell** [comment]
To give a more detailed explanation: the way ratings are stored is through the "rated" folder or whichever folder you have set that folder to be (it can be changed) so if it's not saving when spotify is closed that means songs are not being added to that folder and implies that there must be a problem with the adding to folder API which is why I asked for the js console logs 👍

**Comment by dantdmdarealguy** [comment]
it seems that the "rated" folder does not have anything inside it and that would probably be the main issue
![image](https://github.com/brimell/spicetify-star-ratings/assets/62715519/2babbac1-a259-47f0-be04-4a907cb9ce15)
![image](https://github.com/brimell/spicetify-star-ratings/assets/62715519/66e7cbce-aebf-49aa-b096-ec7e7a87c82c)

**Comment by dantdmdarealguy** [comment]
the issue seems to be on this line
![image](https://github.com/brimell/spicetify-star-ratings/assets/62715519/11841f57-df0a-4378-9f97-06a63859fa10)

**Comment by brimell** [comment]
Ok I'll check it out

**Comment by ThanatosHM** [comment]
I have the same probleme and the "use as a rated folder" don't seem to work.

**Comment by brimell** [comment]
So sorry, I'm super busy right now. I will take a look soon but it's not too much of a problem right as now the default Rated folder sounds like it's working still

**Comment by hiimral** [comment]
bump

**Comment by brimell** [comment]
Yeah so sorry for not being on it but for the next few months still I really don't have any time for side projects 😢. I'll definitely take a look at it after I'm through this busy period but unfortunately that's not any time soon 😭

**Comment by AbanOnPC** [comment]
Same issue here

**Comment by hiimral** [comment]
> Yeah so sorry for not being on it but for the next few months still I really don't have any time for side projects 😢. I'll definitely take a look at it after I'm through this busy period but unfortunately that's not any time soon 😭

thats fine man, take care of urself

**Comment by hiimral** [comment]
you can fix this issue by manually creating playlist in the "Rated" folder if anyone is wondering, thank you for making the extension!

**Comment by brimell** [comment]
ah after looking at it for longer I think I've realised the problem
the api to get the user's username has been changed (spotify change stuff like this all the time which is really annoying) so I'm just trying to fix it right now

**Comment by Spook2U** [comment]
Any update? still broken for me

**Comment by brimell** [comment]
Yeah so unfortunately for now you're going to have to just create a folder called "Rated" and make the playlist names of 0.0, 0.5, 1.0, 1.5 etc like the following:
![image](https://github.com/user-attachments/assets/9b174d86-c1e9-4d0a-a888-5c4be13b5792)

**Comment by Spook2U** [comment]
Did this, but still getting the error. "Cannot read properties of undefined (reading 'getUsername')

**Comment by Truthlike** [comment]
> Yeah so unfortunately for now you're going to have to just create a folder called "Rated" and make the playlist names of 0.0, 0.5, 1.0, 1.5 etc like the following: ![image](https://github.com/user-attachments/assets/9b174d86-c1e9-4d0a-a888-5c4be13b5792)

I did this and it works

The key is it seems it to ''use as rated folder'' which you can access after right click the rated folder (needs 0.0-5.0)

![Image](https://github.com/user-attachments/assets/c71386c3-1a1e-424b-89ea-085645775669)

hopefully this helps!

btw it will default to adding to liked songs but this seems to redirect it to the ''ratings folder'' (though it still goes in liked songs)

**Comment by Spook2U** [comment]
Ah finally! I have created the folder already, I missed that you need to right click and set it as rating folder. 
Now it works.

**Comment by riotinprogress** [comment]
Truthlike's method is indeed working for me. Fresh install. I've been on Spotify's forums dating back over a decade asking for star ratings and personal playcounts in order to make proper smart playlists (I've asked for that as well.) 

They have zero intention of adding these features in order to make Spotify an actual music library player.

**Comment by jullanggit** [comment]
Is this still an issue?

**Comment by jullanggit** [comment]
This should no longer be an issue, please re-open it it still is.
